packages/python/patch-pthread_key_t

```diff
@@ -0,0 +1,269 @@
+ void
+-PyThread_delete_key(int key)
++PyThread_delete_key(PyThread_type_key key)
+ {
+     struct key *p, **q;
+ 
+@@ -283,7 +283,7 @@
+ }
+ 
+ int
+-PyThread_set_key_value(int key, void *value)
++PyThread_set_key_value(PyThread_type_key key, void *value)
+ {
+     struct key *p;
+ 
+@@ -298,7 +298,7 @@
+  * if the current thread doesn't have an association for key.
+  */
+ void *
+-PyThread_get_key_value(int key)
++PyThread_get_key_value(PyThread_type_key key)
+ {
+     struct key *p = find_key(0, key, NULL);
+ 
+@@ -310,7 +310,7 @@
+ 
+ /* Forget the current thread's association for key, if any. */
+ void
+-PyThread_delete_key_value(int key)
++PyThread_delete_key_value(PyThread_type_key key)
+ {
+     long id = PyThread_get_thread_ident();
+     struct key *p, **q;
+--- Python/thread_foobar.h
++++ Python/thread_foobar.h
+@@ -85,7 +85,7 @@
+ #define Py_HAVE_NATIVE_TLS
+ 
+ #ifdef Py_HAVE_NATIVE_TLS
+-int
++PyThread_type_key
+ PyThread_create_key(void)
+ {
+     int result;
+@@ -93,13 +93,13 @@
+ }
+ 
+ void
+-PyThread_delete_key(int key)
++PyThread_delete_key(PyThread_type_key key)
+ {
+ 
+ }
+ 
+ int
+-PyThread_set_key_value(int key, void *value)
++PyThread_set_key_value(PyThread_type_key key, void *value)
+ {
+     int ok;
+ 
+@@ -110,7 +110,7 @@
+ }
+ 
+ void *
+-PyThread_get_key_value(int key)
++PyThread_get_key_value(PyThread_type_key key)
+ {
+     void *result;
+ 
+@@ -118,7 +118,7 @@
+ }
+ 
+ void
+-PyThread_delete_key_value(int key)
++PyThread_delete_key_value(PyThread_type_key key)
+ {
+ 
+ }
+--- Python/thread_nt.h
++++ Python/thread_nt.h
+@@ -374,23 +374,23 @@
+ #define Py_HAVE_NATIVE_TLS
+ 
+ #ifdef Py_HAVE_NATIVE_TLS
+-int
++PyThread_type_key
+ PyThread_create_key(void)
+ {
+     DWORD result= TlsAlloc();
+     if (result == TLS_OUT_OF_INDEXES)
+         return -1;
+-    return (int)result;
++    return (PyThread_type_key)result;
+ }
+ 
+ void
+-PyThread_delete_key(int key)
++PyThread_delete_key(PyThread_type_key key)
+ {
+     TlsFree(key);
+ }
+ 
+ int
+-PyThread_set_key_value(int key, void *value)
++PyThread_set_key_value(PyThread_type_key key, void *value)
+ {
+     BOOL ok;
+ 
+@@ -401,7 +401,7 @@
+ }
+ 
+ void *
+-PyThread_get_key_value(int key)
++PyThread_get_key_value(PyThread_type_key key)
+ {
+     /* because TLS is used in the Py_END_ALLOW_THREAD macro,
+      * it is necessary to preserve the windows error state, because
+@@ -416,7 +416,7 @@
+ }
+ 
+ void
+-PyThread_delete_key_value(int key)
++PyThread_delete_key_value(PyThread_type_key key)
+ {
+     /* NULL is used as "key missing", and it is also the default
+      * given by TlsGetValue() if nothing has been set yet.
+--- Python/thread_pthread.h
++++ Python/thread_pthread.h
+@@ -603,46 +603,66 @@
+ 
+ #define Py_HAVE_NATIVE_TLS
+ 
+-int
++PyThread_type_key
+ PyThread_create_key(void)
+ {
+     pthread_key_t key;
+     int fail = pthread_key_create(&key, NULL);
+     if (fail)
+         return -1;
++#if !defined(__CloudABI__)
+     if (key > INT_MAX) {
```

Comment:
I've removed the pthread_type_key patch and created https://github.com/moreati/cloudabi-ports/issues/5 to record the lack of threading support